### PR TITLE
[EP-2486] Prevent invalid payment methods from being chosen

### PR DESCRIPTION
**Pull request by canac**
Prevent invalid payment methods from being selected by default on step 2 of checkout.

When testing locally, I used the following change to simulate invalid payment methods:

```patch
diff --git a/src/common/services/api/order.service.js b/src/common/services/api/order.service.js
index fcbb2ad4..475cce6a 100644
--- a/src/common/services/api/order.service.js
+++ b/src/common/services/api/order.service.js
@@ -234,6 +234,7 @@ class Order {
           paymentMethods.unshift(selector.chosen)
         }
         return map(paymentMethods, paymentMethod => {
+          paymentMethod.description['payment-instrument-identification-attributes']['expiry-year'] = '2020'
           paymentMethod.description.selectAction = paymentMethod.self.uri
           if (paymentMethod.description['payment-instrument-identification-attributes']['street-address']) {
             paymentMethod.description.address =
```

https://jira.cru.org/browse/EP-2486

### CODE REVIEW

**Review by wrandall22 — APPROVED** [review]

**Review by dr-bizz — APPROVED** [review]
The code looks good. Great work. Can you merge it onto staging so I can test it there?

**Review by dr-bizz — COMMENTED** [review]
Might be good to add a test to check `selectedPaymentMethod` is `null` when all cards are invalid.

**Comment by canac** [comment]
@dr-bizz I added that test and merged into staging.

**Comment by wrandall22** [comment]
So this will prevent default selection; just wanted to make sure it will also prevent selecting a payment method with an invalid date? They should be forced to update the expiration date if they choose a payment method with an expired date.

**Comment by canac** [comment]
@wrandall22 Correct, invalid payment methods have always been disabled https://github.com/CruGlobal/give-web/blob/27de8baafe7080a1211c5efeccecac07cd30e49f/src/app/checkout/step-2/existingPaymentMethods/existingPaymentMethods.tpl.html#L7-L13 (note the `ng-init="expired = !$ctrl.validPaymentMethod(paymentMethod)"` on line 7 and the `ng-disabled="expired"` on line 9) The bug was that the previous code could initially set `$ctrl.selectedPaymentMethod` to an invalid payment method.

**Comment by dr-bizz** [comment]
Looks good in staging

**Comment by dr-bizz** [comment]
We should have done more testing with this PR. Part of that is on me. Anything with checkout should be tripled-checked before rolling into production.
